tools/autograd/utils.py

```diff
@@ -47,6 +47,9 @@ def split_name_params(prototype):
 def uninplace_api_name(api_name):
     if api_name.endswith('_') and not api_name.endswith('__'):
         api_name = api_name[:-1]
+    return unout_api_name(api_name)
+
+def unout_api_name(api_name):
```

Comment:
blegh

Reply:
agreed. I'm only ok with this because it's the legacy codegen.